frameworks/Java/armeria/src/main/java/hello/services/PostgresDbService.java

```diff
@@ -36,37 +32,26 @@ public PostgresDbService() {
   }
 
   @Get("/db")
-  public HttpResponse db() throws Exception {
-    return HttpResponse.of(
-    	HttpHeadersHelper.getHttpHeader(MediaType.JSON),
-        HttpData.of(MAPPER.writeValueAsBytes(getWorld(getRandomNumber())))
-        );
+  @ProducesJson
+  public CompletableFuture<World> db(ServiceRequestContext ctx) throws Exception {
+    return getWorld(getRandomNumber(), ctx);
   }
 
   // need to use regex as /queries/{count} doesn't work when count is null
   @Get("regex:^/queries/(?<count>.*)$")
```

Comment:
How about splitting this into `@Get("/queries/")` and `@Get("/queries/{count}")` ?